client/tests/integration/account_registration/public.rs

```diff
@@ -0,0 +1,102 @@
+//! In public use cases, a new account should be:
+//! - recognized when targeted as (a part of) a destination (or an object) of a creative instruction
+//!     - becomes able to hold assets, permissions, roles, and metadata
+//!     - cannot yet execute any queries or transactions
+//! - automatically activated by an administrative trigger
+//!     - becomes able to execute queries and transactions
+
+use eyre::Result;
+use iroha_client::data_model::prelude::*;
+use test_network::*;
+use test_samples::{gen_account_in, ALICE_ID};
+
+/// A new account e.g. "carol" should be:
+/// - recognized when targeted as a destination of a transfer of numeric asset e.g. "rose"
+/// - automatically activated by an administrative trigger
+///
+/// # Scenario
+///
+/// 0. ... administrative trigger registered
+/// 0. new carol targeted ... carol recognized
+/// 0. clock forward by one block ... carol activated
+/// 0. carol tries query ... ok
+/// 0. carol tries transaction ... ok
+#[test]
+fn on_transfer_asset_numeric() -> Result<()> {
+    let (_rt, _peer, client_alice) = <PeerBuilder>::new().with_port(10_810).start_with_runtime();
+    wait_for_genesis_committed(&[client_alice.clone()], 0);
+    let observer = client_alice.clone();
+
+    // ... administrative trigger registered
+    let wasm = iroha_wasm_builder::Builder::new(
+        "tests/integration/smartcontracts/trigger_activate_account",
+    )
+    .show_output()
+    .build()?
+    .optimize()?
+    .into_bytes()?;
+    let register_activator = Register::trigger(Trigger::new(
+        "activator$wonderland".parse().unwrap(),
+        Action::new(
+            WasmSmartContract::from_compiled(wasm),
+            Repeats::Indefinitely,
+            ALICE_ID.clone(),
```

Comment:
The trigger authority is tentatively set to alice, but once the account ID is expanded to off-curve multihash, this should be a system (non-personal) authority

Reply:
Is there an issue for it?

Reply:
I don't think it has been broken down into issues yet.
If we want to support use cases where a trigger needs different permission than the registrant person (e.g. swap between two accounts, #1212), the trigger should reference its own account (or should be identified as an account).
The idea of _system level triggers_ and _technical accounts_ was suggested in as early as [Triggers ADR](https://wiki.hyperledger.org/display/iroha/Triggers).
Similar concepts include [Contract Address](https://info.etherscan.com/what-is-an-ethereum-address/) and [Program Derived Addresses](https://solanacookbook.com/core-concepts/pdas.html)

Reply:
I found my first comment off topic. In this use case the trigger manages domain membership, so alice (domain owner) would be an appropriate authority

Reply:
- created #4670